src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,148 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> tail;
+    private Node<T> head;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
```

Comment:
```suggestion
        Node<T> newNode = new Node<>(tail, value, null);
```

Reply:
Thanks for the mentorship. It's really interesting how the code can be improved.